presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -273,6 +267,29 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
         return concat(partitionBatches);
     }
 
+    private void ensureBucketingMatch(HiveBucketProperty tableBucketProperty, HivePartition hivePartition, Partition partition)
+    {
+        Optional<HiveBucketProperty> optionalPartitionBucketProperty = partition.getStorage().getBucketProperty();
```

Comment:
What about `checkCondition(...)` to ensure that `optionalPartitionBucketProperty` is non-empty instead of this big if-else block below?
E.g.:
```java
checkCondition(optionalPartitionBucketProperty.isPresent(), ...); // error message from else block below

HiveBucketProperty partitionBucketProperty = ...;
...
```